### ISSUE TITLE
Change url based on environemtn

### ISSUE THREAD

**Issue by ianwremmel** [Netflix/pollyjs#337]
## Description

I _think_ this is just a documentation issue; what I'm trying to do appears to be possible, but I can't seem to get it working. 

I'm using dynamdb-local via docker throughout my test suite. On my laptop, dynamodb is available at http://localhost:8080; in my CI environemnt, dynamodb is available at http://dynamodb:8000, which is set via an CI-wide environment variable, `AWS_ENDPOINT`. All of the code under test reads `AWS_ENDPOINT` to determine what to talk to and, as a result, the recordings made locally don't match up with the urls that are requested in CI.

It seems like I should able to use `config.url.hostname()` and `config.url.port()` to work around this mismatch, but so far, I haven't been able to find any combination that actually works. 

Is this something that should be possible?

If so, could you provide more specific documentation on how to use the `url` config to solve this kind of situation?

### Error Message & Stack Trace

```
COPY THE ERROR MESSAGE, INCLUDING STACK TRACE HERE
```

### Config

Copy the config used to setup the Polly instance:

```js
      url: {
        hostname(hostname, req) {
          if (hostname === 'dynamodb') {
            return 'localhost';
          }
          return hostname;
        },
        port(port, req) {
          if (Number(port) === 8000) {
            return 8080;
          }
          return port;
        },
      },
```

### Dependencies

Copy the @pollyjs dependencies from `package.json`:

```json
{
    "@pollyjs/adapter-node-http": "^4.2.1",
    "@pollyjs/core": "^4.2.1",
    "@pollyjs/persister-fs": "^4.2.1",
}
```

## Environment

```
Node.js v12.16.1
darwin 19.4.0
6.13.4
zsh: command not found: yarn
```

**Comment by offirgolan** [comment]
@ianwremmel your configuration looks correct. When you turn on logging, you should be able to see the `identifiers` object for each request that the id gets generated from. Does it look right for both environments?

**Comment by ianwremmel** [comment]
Sorry for the slow response here. it's taken a while to get back to the project. I hope to have an answer for you soonish.

**Comment by ianwremmel** [comment]
I think I know what happened here. I haven't been able to test in that project yet, but I ran into a similar issue adding polly to a new project. In the new project, I've only used polly locally so far, but it kept failing to match. I'm using supertest and an express app, so I was getting a new port on every test running, thus preventing matches. 

I tried 

```
url: {
  port: false,
},
```

but the problem continued. I had to add 

```
headers: {
  exclude ['host']
}
```

to get things working. I'm guessing I was running into a similar issue with the `dynamodb`/`localhost` situation: even though I was adjusting the hostname polly was using, the `host` header still ended up in the identifiers list with `localhost` instead of `dynamodb`.
